Lib/axis.py

```diff
@@ -2192,6 +2192,8 @@ def __setitem__(self, index, value):
 
     def __setslice__(self, low, high, value):
         # Hack to prevent netCDF overflow error on 64-bit architectures
+        if(self.isUnlimited() and self.isTime() and (high >= Max32int)):
```

Comment:
I guess I can remove the isTime, usually the `Unlimited` axis is for time, but the issue will come back for all axis.  

I used `>= Max32int` for it is the test on the next line.  I was thinking that if someone want to add some data  let says in your example `t[1:121]=100.0`,  the `121` will be overwritten to `120`. 

 Actually if someone does `t[121:121]=100.0` that would be change to `t[121:120]=100.0` and the new slice will raise an exception.